soccer/ui/MainWindow.cpp

```diff
@@ -306,7 +314,7 @@ void MainWindow::updateViews() {
 
     GameState game_state;
     {
-        std::lock_guard<std::mutex> lock(_context_mutex);
+        std::lock_guard<std::mutex> lock(*_context_mutex);
         game_state = _context->game_state;
     }
```

Comment:
Can we not have to wait on this mutex? We sometimes spend a lot of time waiting on this mutex here. Maybe use a SPSC queue or something?

Reply:
We spend a lot of time waiting on this mutex because we have to wait for processor to finish its cycle. However, this shouldn't bring frame rate down substantially below processor's frame rates on its own.